dsl/jpql/src/main/kotlin/com/linecorp/kotlinjdsl/dsl/jpql/Jpql.kt

```diff
@@ -536,6 +536,54 @@ open class Jpql : JpqlDsl {
         return Expressions.div(this.toExpression(), value.toExpression())
     }
 
+    /**
+     * Creates an expression that represents the modulo of values.
+     *
+     * This is the same as ```(value1) % (value2)```.
+     */
+    @JvmName("modWithParentheses")
+    @SinceJdsl("3.4.0")
+    fun <T : Number, S : T?> mod(value1: Expressionable<@Exact T>, value2: S): Expression<T> {
+        return Expressions.mod(
+            Expressions.parentheses(value1.toExpression()),
+            Expressions.parentheses(Expressions.value(value2)),
+        )
+    }
+
+    /**
+     * Creates an expression that represents the modulo of values.
+     *
+     * This is the same as ```(value1) % (value2)```.
+     */
+    @JvmName("modWithParentheses")
+    @SinceJdsl("3.4.0")
+    fun <T : Number, S : T> mod(value1: Expressionable<@Exact T>, value2: Expressionable<S>): Expression<T> {
+        return Expressions.mod(
+            Expressions.parentheses(value1.toExpression()),
+            Expressions.parentheses(value2.toExpression()),
+        )
+    }
+
+    /**
+     * Creates an expression that represents the modulo of values.
+     *
+     * This is the same as ```(value1) % (value2)```.
+     */
+    @SinceJdsl("3.4.0")
+    fun <T : Number, S : T?> Expressionable<@Exact T>.mod(value: S): Expression<T> {
```

Comment:
[In JPQL](https://jakarta.ee/specifications/persistence/3.1/jakarta-persistence-spec-3.1.html#a5284), mod is not an operator, so I think it would be better to provide it as a function to reduce confusion.

So, unfortunately, I'd like to ask you to remove the extension function and modify the comment from `value1 % value2` to `mod(value1, value2)`

Reply:
Also, in the [Jakarta specification](https://jakarta.ee/specifications/persistence/3.1/jakarta-persistence-spec-3.1.html#a5284), the MOD can only accept an integer. Please modify it to accept only integers and not numbers.